### PR TITLE
if NOSCRIPT occurs, reload the script

### PR DESCRIPTION
**Pull request by phutwo**
When Redis fails over or reboots, the scripts may be lost, since we've already
loaded them in local memory, we assume the script is already there. In the event
that they're no longer there, reload from memory and retry once

### CODE REVIEW

File: rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java [hunk 1]
```diff
@@ -60,6 +62,24 @@ private Object callEvalDynamicList(MethodContext context, Object[] objects) {
...
 
-        return  jedis.evalsha(context.getSha1(), keys, argv);
+        return  evalShaHandleReloadScript(context, keys, argv);
+    }
+
+    private Object evalShaHandleReloadScript(MethodContext context, List<String> keys, List<String> argv) {
+        try {
+            return jedis.evalsha(context.getSha1(), keys, argv);
+        } catch (JedisDataException e) {
+            if (e.getMessage().contains("NOSCRIPT")) {
```
Comment (matthewbogner) [inline rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java]: potential NPE if the message is null.

File: rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java [hunk 2]
```diff
@@ -60,6 +62,24 @@ private Object callEvalDynamicList(MethodContext context, Object[] objects) {
...
+                //If it throws again, we can back-off or we can just let it throw again. In this case, I think we should
+                //let it throw because most likely will be trying the same thing again and hopefully it will succeed later.
+                final String sha2 = jedis.scriptLoad(context.getLuaContext().getRenderedLuaString());
```
Comment (matthewbogner) [inline rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java]: being that sha2 is a legitimate hashing algorithm, I would recommend choosing an alternate variable name.

File: rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java (hunk 1 above)
Comment (phutwo) [inline rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java]: done

File: rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java (hunk 2 above)
Comment (phutwo) [inline rdbi-core/src/main/java/com/lithium/dbi/rdbi/MethodContextInterceptor.java]: done

**Pull request merged**